src/pages/vendors/create/Create.tsx

```diff
@@ -54,36 +56,27 @@ export default function Create() {
   const [vendor, setVendor] = useState<Vendor>();
   const [errors, setErrors] = useState<ValidationBag>();
 
+  const [contacts, setContacts] = useState<Partial<VendorContact>[]>([
+    {
+      first_name: '',
+      last_name: '',
+      email: '',
+      phone: '',
+      send_email: false,
+    },
+  ]);
+
   useEffect(() => {
     if (data) {
       setVendor({
         ...data,
         country_id: '',
-        contacts: [
-          {
-            id: '',
-            first_name: '',
-            last_name: '',
-            email: '',
-            send_email: true,
-            created_at: 0,
-            updated_at: 0,
-            archived_at: 0,
-            is_primary: false,
-            phone: '',
-            custom_value1: '',
-            custom_value2: '',
-            custom_value3: '',
-            custom_value4: '',
-            link: '',
-            last_login: 0,
-          },
-        ],
```

Comment:
Why are we decoupling the contacts from the main object?

Reply:
@beganovich Hmm, for the logic regarding Vendor Contacts, I borrowed the logic from the Client Contacts. In fact, we have a state for contacts that handles the entire logic. Then, on save, we simply transfer what we have in the contacts state into the main object. Both approaches look similar, but I adopted this one from the Client Contacts. Perhaps you'll prefer this one more. However, to avoid any potential edge cases that could break functionality, I've removed it. The logic should reside either in the contacts state or in the main object. Please let me know which approach you prefer.